ext/ffi/ffi.c

```diff
@@ -339,6 +340,9 @@ static ffi_type *zend_ffi_make_fake_struct_type(zend_ffi_type *type) /* {{{ */
 			case ZEND_FFI_TYPE_POINTER:
 				t->elements[i] = &ffi_type_pointer;
 				break;
+            case ZEND_FFI_TYPE_STRUCT:
+                t->elements[i] = &ffi_type_pointer;
```

Comment:
I'm not 100% sure why this works tbh. In my test's specific case this makes sure that the returned value won't be NULL. But I failed to trace back how `ffi_type_pointer` is actually used in the code.

I suppose `ffi_type_*` symbols come from libffi and given `ffi_type_struct` does not exist I just guessed that `ffi_type_pointer` would be the best way to go. Please let me know if this isn't a good idea.

Reply:
I doubt this is the right thing to do here. Probably only works because in your test case the struct happens to have the same size as a pointer.

Reply:
The fix is incorrect. We can't pass/return structures by value in general. libffi doesn't support this (e.g. see https://github.com/libffi/libffi/issues/33). However, we may try to pass/return structures by value if they fit into 32/64-bit word. I'll try to do this...

Reply:
I was wrong. libffi seems support big structures. See my fix https://github.com/php/php-src/pull/6781

Reply:
Nice! Thanks a lot for taking care of this @dstogov. Hopefully next time I'll manage it by myself, but I'm very glad learning from your fix :)

Shall we close this pull request and follow up on #6781 ?